eng/testing/tests.targets

```diff
@@ -72,10 +72,7 @@
   <ItemGroup Condition="'$(IsXUnitLogCheckerSupported)' == 'true' and
                         '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and
                          $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), '$(NETCoreAppCurrentVersion)'))">
-    <SetScriptCommands Condition="'$(TargetOS)' == 'windows'" Include="set __TestArchitecture=$(TargetArchitecture)" />
     <SetScriptCommands Condition="'$(TargetOS)' == 'windows'" Include="set __IsXUnitLogCheckerSupported=1" />
```

Comment:
Is XUnitLogChecker just a CI tool or does it work locally as well when running tests? If the former then it might make sense to condition this item group on `'$(ArchiveTests)' == 'true'`.

Reply:
Maybe @ivdiazsa or @carlossanlop can answer that.

I realized I didn't want to mess with the XUnitLogChecker since a recent fix made it not break on local dev machines. So I've scoped back this PR to only fix the architecture-specific setting of `__TestArchitecture`.

Reply:
XUnitLogChecker can be run locally, but it has to be called by the user. Automatically, it's only run through the CI scripts.